File: src/Tests/Behavioral/ChannelReceiveFromClosed/ChannelReceivedFromClosed.go
package main

import "fmt"

func main() {
        	c := make(chan int, 3)
	        c <- 1
        	c <- 2
	        c <- 3
	        close(c)
	        for i := 0; i < 4; i++ {
		                fmt.Printf("%d ", <-c) // prints 1 2 3 0
	        }
}

File: src/Tests/Behavioral/ChannelReceiveFromClosed/ChannelReceivedFromClosed.cs
namespace go;

using fmt = fmt_package;

partial class main_package {

private static void Main() {
    var c = new channel<nint>(3);
    c.ᐸꟷ(1);
    c.ᐸꟷ(2);
    c.ᐸꟷ(3);
    close(c);
    for (nint i = 0; i < 4; i++) {
        fmt.Printf("%d "u8, ᐸꟷ(c));
    }
}

} // end main_package
